controller/update.go

```diff
@@ -72,47 +72,37 @@ func (c controller) getNumRunningSlices(ctx context.Context, req Request) (int,
 	return len(sliceIDs), nil
 }
 
-func (c controller) isGroupRemovalAllowed(ctx context.Context, req Request, minAlive int) (bool, error) {
+func (c controller) isGroupRemovalAllowed(ctx context.Context, req Request, minAlive int, removalInProgress *int64) (bool, error) {
 	c.Config.Logger.Debug(ctx, "controller: checking group removal allowed, req: %v", req)
 
-	numRunning, err := c.getNumRunningSlices(ctx, req)
-	if err != nil {
-		return false, maskAny(err)
-	}
-
 	c.Config.Logger.Debug(
-		ctx, "controller: numRunning: %v, minAlive: %v",
-		numRunning, minAlive,
+		ctx, "controller: removalInProgress: %v, minAlive: %v",
+		*removalInProgress, minAlive,
 	)
 
-	if numRunning > minAlive {
-		c.Config.Logger.Debug(ctx, "controller: group removal allowed (numRunning > minAlive)")
+	if (minAlive - int(*removalInProgress)) > 0 {
+		c.Config.Logger.Debug(ctx, "controller: group removal allowed ((minAlive - int(removalInProgress)) > 0)")
 		return true, nil
 	}
 
-	c.Config.Logger.Debug(ctx, "controller: group removal not allowed (numRunning <= minAlive)")
+	c.Config.Logger.Debug(ctx, "controller: group removal not allowed ((minAlive - int(removalInProgress)) <= 0)")
 	return false, nil
```

Comment:
I'd like to keep it in, as it was useful while writing the fix. I guess it could come in handy at some point in the future